src/submission/repositories/xpub-submission-root.test.ts

```diff
@@ -34,32 +34,28 @@ const testDatabaseEntry2 = {
 const databaseEntries = [testDatabaseEntry, testDatabaseEntry2];
 
 describe('Knex Submission Repository', () => {
-    let mockKnex: MockKnex;
+    let adapter: KnexTableAdapter;
+    let mock: MockKnex;
 
     beforeEach(() => {
         jest.resetAllMocks();
-        mockKnex = new MockKnex();
-        mockKnex.insert = jest.fn().mockReturnValue(mockKnex);
-        mockKnex.withSchema = jest.fn().mockReturnValue(mockKnex);
-        mockKnex.into = jest.fn().mockReturnValue(mockKnex);
-        mockKnex.update = jest.fn().mockReturnValue(mockKnex);
-        mockKnex.select = jest.fn().mockReturnValue(mockKnex);
-        mockKnex.from = jest.fn().mockReturnValue(mockKnex);
-        mockKnex.where = jest.fn().mockReturnValue(mockKnex);
-        mockKnex.returning = jest.fn().mockReturnValue(mockKnex);
+        mock = new MockKnex();
+        adapter = createMockAdapter(mock);
     });
 
     describe('findAll', () => {
         it('returns the correct number of entries', async (): Promise<void> => {
-            mockKnex.from = jest.fn().mockReturnValue(databaseEntries);
-            const repo = new XpubSubmissionRootRepository((mockKnex as unknown) as Knex);
+            adapter.executor = jest.fn().mockReturnValue(databaseEntries);
+            const repo = new XpubSubmissionRootRepository(adapter);
             const result = await repo.findAll();
             expect(result).toHaveLength(2);
         });
         it('returns a converted DTO object from the database entries', async (): Promise<void> => {
-            mockKnex.from = jest.fn().mockReturnValue(databaseEntries);
-            const repo = new XpubSubmissionRootRepository((mockKnex as unknown) as Knex);
+            adapter.executor = jest.fn().mockReturnValue(databaseEntries);
+            const repo = new XpubSubmissionRootRepository(adapter);
             const result = await repo.findAll();
+            expect(mock.select).toBeCalled();
```

Comment:
Do we need to assert underlying knex mock if we are mocking the adapter?

Reply:
So the idea is to ensure we are creating well formed sql ... it was only this morning that I realised that we forgot to include the "from()" in selects - and it would be great to catch if these are missing in the tests - before we start any integration tests or end up in production!